src/main/java/nl/tudelft/semgroup4/collision/DefaultCollisionHandler.java

```diff
@@ -174,15 +191,16 @@ public void run() {
             }, 1000);
         }
     };
-
-    final CollisionHandler<Projectile, Wall> projectileWallHandler = (game, projectile, wall) -> {
+    
+    final CollisionHandler<AbstractWall, Projectile> wallProjectileHandler = 
+            (game, wall, projectile) -> {
         Game.LOGGER.log(LogSeverity.VERBOSE, "Collision", "Projectile - wall collision");
         
         final Shape projectileRect = projectile.getBounds();
         final Shape wallRect = wall.getBounds();
         // This structure makes the projectile ignore any walls below it (such as the floor wall).
 
-        if (wallRect.getY() < projectileRect.getY()) {
+        if (wallRect.getY() < projectileRect.getY() || !(wall instanceof RegularWall)) {
```

Comment:
Is it not better to define two different Collisionhandlers? One for <code>RegularWall, Projectile</code> and one for <code>MovingWall, Projectile</code> or even <code>VerMovingWall, Projectile</code> (I'm imagining that this check is only necessary for downwards-moving walls, thus only for VerMovingWalls)

This avoids the instanceof, in a clear and easy way.

< interface, organisation